components/engine_traits/src/engine.rs

```diff
@@ -20,30 +21,77 @@ pub trait WriteBatch: Mutable {
     fn rollback_to_save_point(&mut self) -> Result<()>;
 }
 
-pub trait KvEngine: Peekable + Mutable + Iterable + Send + Sync + Clone {
+pub trait KvEngine: Peekable + Mutable + Iterable + Send + Sync + Clone + Debug + 'static {
     type Snap: Snapshot;
     type Batch: WriteBatch;
+    type DBOptions: DBOptions;
+    // FIXME: With the current rocks implementation this
+    // type wants to be able to contain a lifetime, but it's
+    // not possible without "generic associated types".
+    //
+    // https://github.com/rust-lang/rfcs/pull/1598
+    // https://github.com/rust-lang/rust/issues/44265
+    type CFHandle: CFHandle;
+    type CFOptions: CFOptions;
+
+    fn get_db_options(&self) -> Self::DBOptions;
+    // FIXME: return type
+    fn set_db_options(&self, options: &[(&str, &str)]) -> StdResult<(), String>;
 
     fn write_opt(&self, opts: &WriteOptions, wb: &Self::Batch) -> Result<()>;
     fn write(&self, wb: &Self::Batch) -> Result<()> {
         self.write_opt(&WriteOptions::default(), wb)
     }
-    fn write_batch(&self, cap: usize) -> Self::Batch;
+    fn write_batch(&self) -> Self::Batch;
+    fn write_batch_with_cap(&self, cap: usize) -> Self::Batch;
     fn snapshot(&self) -> Self::Snap;
     fn sync(&self) -> Result<()>;
+
     fn cf_names(&self) -> Vec<&str>;
+    fn cf_handle(&self, name: &str) -> Option<&Self::CFHandle>;
+    fn get_options_cf(&self, cf: &Self::CFHandle) -> Self::CFOptions;
+    // FIXME: return type
+    fn set_options_cf(
+        &self,
+        cf: &Self::CFHandle,
+        options: &[(&str, &str)],
+    ) -> StdResult<(), String>;
+
     fn delete_all_in_range(&self, start_key: &[u8], end_key: &[u8]) -> Result<()> {
         if start_key >= end_key {
             return Ok(());
         }
         for cf in self.cf_names() {
-            self.delete_all_in_range_cf(cf, start_key, end_key)?;
+            self.delete_all_in_range_cf(cf, start_key, end_key, false)?;
         }
         Ok(())
     }
-    fn delete_all_in_range_cf(&self, cf: &str, start_key: &[u8], end_key: &[u8]) -> Result<()>;
+    fn delete_all_in_range_cf(
+        &self,
+        cf: &str,
+        start_key: &[u8],
+        end_key: &[u8],
+        use_delete_range: bool,
+    ) -> Result<()>;
+    fn delete_files_in_range_cf(
```

Comment:
why add this?

Reply:
I don't see any reason for it. I'll remove it.